apps/test/unit/code-studio/components/progress/LessonLockDialogTest.jsx

```diff
@@ -1,112 +1,188 @@
 import React from 'react';
-import {shallow} from 'enzyme';
+import {mount} from 'enzyme';
 import {expect} from '../../../../util/reconfiguredChai';
 import sinon from 'sinon';
 import {UnconnectedLessonLockDialog as LessonLockDialog} from '@cdo/apps/code-studio/components/progress/LessonLockDialog';
 import {LockStatus} from '@cdo/apps/code-studio/lessonLockRedux';
 
+// This * import allows us to stub out the SectionSelector component
+import * as SectionSelector from '@cdo/apps/code-studio/components/progress/SectionSelector';
+
+const fakeSectionId = 42;
+
 const MINIMUM_PROPS = {
-  isOpen: false,
+  isOpen: true,
   handleClose: () => {},
   initialLockStatus: [],
-  selectedSectionId: null,
+  selectedSectionId: fakeSectionId,
   saving: false,
   saveDialog: () => {}
 };
 
-const fakeSectionId = 1;
+// Helper function to get the list of rows in the student table.
+const getStudentRows = wrapper =>
+  wrapper.find('#ui-test-student-table tbody tr');
+
+// Helper function to verify that the radio buttons in the given student row
+// correctly reflect the expected lock status.
+const verifyRowLockStatus = (row, expectedLockStatus) => {
```

Comment:
Love this use of a helper function here. Could you add to the comment a pseudo- diagram of what the different buttons are so it's a little easier to visualize? (aka something like [ ] Locked | [ ] Editable | [ ] Read-Only)

Reply:
Good idea, will add.

Reply:
If you had a separate component for renderStudentRow, you could potentially just test that the expected props are sent to that component, which could simplify these tests a little bit

Reply:
This is an interesting side discussion too -- there seems to be some debate currently on whether shallow rendering (and verifying props as you suggest) or deep rendering is the preferred approach to unit testing react components.  From what I read, it seems like the winds are blowing towards [deep rendering](https://kentcdodds.com/blog/why-i-never-use-shallow-rendering) at the moment which was part of the reason I went this approach here.  I'd love to discuss this more with a broader group and see if we can form a shared direction here.

Reply:
The conversations I've had here in the past have strongly preferred using `shallow` if possible and `mount` only when necessary. @davidsbailey wrote up a [doc](https://docs.google.com/document/d/1cLcTvqh-F65IhEMlI6wVEQVDImUwoKhYhEv_O71EML8/edit#heading=h.od2d1042x84v) about a year ago about this and I know that's what I've been trying to adhere to.

I don't know what the future holds here for the industry as a whole, but wanted to add that data point for our team.

Reply:
Oh, thanks for the pointer, I'd forgotten about that doc.  I left a [comment](https://docs.google.com/document/d/1cLcTvqh-F65IhEMlI6wVEQVDImUwoKhYhEv_O71EML8/edit?disco=AAAAXU23zcg) there to see if this is something we want to open up for discussion again.  In the meantime, I will try to use shallow rendering where possible (but I still prefer to not mix the two in the same test file).